clients/extension/scripts/popup.js

```diff
@@ -176,13 +177,13 @@ async function onGenerateReportButtonClick(background, settings) {
     // Close popup once report is opened in a new tab
     window.close();
   } catch (err) {
-    let message = err.message;
+    let message = err.friendlyMessage || err.message;
     let includeReportLink = true;
 
     // Check for errors in how the user ran Lighthouse and replace with a more
     // helpful message (and remove 'Report Error' link).
     for (const [test, replacement] of Object.entries(NON_BUG_ERROR_MESSAGES)) {
-      if (message.includes(test)) {
+      if (err.message.includes(test)) {
```

Comment:
I think some of our tests actually test our friendly message, maybe we should fail on either?

Reply:
Oh, it's to replace messages to something more descriptive and remove the button as it's more or less a user issue or a chrome issue.

```
const NON_BUG_ERROR_MESSAGES = {
  'Another debugger': 'You probably have DevTools open. Close DevTools to use Lighthouse',
  'multiple tabs': 'You probably have multiple tabs open to the same origin. ' +
      'Close the other tabs to use Lighthouse.',
  // The extension debugger API is forbidden from attaching to the web store.
  // @see https://chromium.googlesource.com/chromium/src/+/5d1f214db0f7996f3c17cd87093d439ce4c7f8f1/chrome/common/extensions/chrome_extensions_client.cc#232
  'The extensions gallery cannot be scripted': 'The Lighthouse extension cannot audit the ' +
      'Chrome Web Store. If necessary, use the Lighthouse CLI to do so.',
  'Cannot access a chrome': 'The Lighthouse extension cannot audit ' +
      'Chrome-specific urls. If necessary, use the Lighthouse CLI to do so.',
  // The user tries to review an error page or has network issues
  'Unable to load the page': 'Unable to load the page. Please verify the url you ' +
      'are trying to review.',
  'Cannot access contents of the page': 'Lighthouse can only audit URLs that start' +
      ' with http:// or https://.',
  'INSECURE_DOCUMENT_REQUEST': 'The URL you have provided does not have valid' +
      ' security credentials.',
  'INVALID_URL': 'Lighthouse can only audit URLs that start' +
      ' with http:// or https://.',
};
```

I can also check for for friendly but don't think we need it at this point but doesn't harm

Reply:
Oh yeah I meant here that some of our string includes I think are looking for strings that are only in the friendly message, so we might want to do something like `err.message.includes(test) || err.friendlyMessage.includes(test)`. 

I just checked all of them though, and I don't think this is actually the case. Most of them are direct from Chrome/extension-specific, some look old and no longer applicable.

Even though we are never showing the file bug button anymore I still think some of the instructions here are worth keeping 👍 

Proposal:

- Replace 'Unable to load the page' test with 'ERRORED_DOCUMENT_REQUEST' and 'FAILED_DOCUMENT_REQUEST' and 'DNS_FAILURE'
- Reorder so 'multiple tabs' and our 'ERROR_CODE*' tests are first and add a comment that the rest are issued by chrome extension API.
- Reword 'INSECURE_DOCUMENT_REQUEST' to say 'valid SSL certificate'?

I know these are only partially related to the change but I'm glad we uncovered them! :D

Reply:
- [x] "Replace 'Unable to load the page' test with 'ERRORED_DOCUMENT_REQUEST' and 'FAILED_DOCUMENT_REQUEST' and 'DNS_FAILURE'"
- [x] Reorder so 'multiple tabs' and our 'ERROR_CODE*' tests are first and add a comment that the rest are issued by chrome extension API.
  * I did it the way around, put the regular errors at the top and chrome stuff at the bottom
- [x] Reword 'INSECURE_DOCUMENT_REQUEST' to say 'valid SSL certificate'?
  * Did it on both places inside lh-errors and extension.